src/message-backup/BackupClient.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Where message backups should be stored
+ */
+export enum BackupType {
+  none,
+  xmtp,
+}
+export interface BackupProvider {
+  type: BackupType
+}
+export type SelectBackupProvider = () => Promise<BackupProvider>
+
+export interface BackupConfiguration {
+  provider: BackupProvider
+  // For the XMTP BackupProvider, this is the name of a topic
+  location?: string
+  // For the XMTP BackupProvider, this is the symmetric encryption key used to encrypt/decrypt backups
+  secret?: Uint8Array
+}
```

Comment:
Is this intention here that all backupConfigurations will follow this format? or will the configuration format be dependent upon the provider?

Reply:
If the Backup configuration is going to be persisted to the network, we'll likely want an associated version so the configuration could be self describing.